sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/CatalogFileIndex.scala

```diff
@@ -94,6 +93,19 @@ class CatalogFileIndex(
     }
   }
 
+  private def listPartitionsByFilter(
+      table: CatalogTable,
+      partitionFilters: Seq[Expression]): Seq[CatalogTablePartition] = {
+    if (sparkSession.sessionState.conf.metastorePartitionPruning) {
+      sparkSession.sessionState.catalog.listPartitionsByFilter(
+        table.identifier, partitionFilters)
+    } else {
+      ExternalCatalogUtils.prunePartitionsByFilter(table,
+        sparkSession.sessionState.catalog.listPartitions(table.identifier),
+        partitionFilters, sparkSession.sessionState.conf.sessionLocalTimeZone)
+    }
+  }
```

Comment:
Could you move this function to `ExternalCatalogUtils` like `prunePartitionsByFilter`?

Reply:
Good suggestion. I tried this at the beginning but stopped since `SparkSession` was not visible in `catalyst` module, but seems we can just pass the `SessionCatalog` as method parameter instead.